### PR TITLE
Do not wait for still grub screens in advent

### PR DESCRIPTION
**Pull request by okurz**
Fixes problems as in https://openqa.opensuse.org/tests/315091

### CODE REVIEW

File: lib/utils.pm [hunk 1]
```diff
@@ -254,7 +254,18 @@ sub type_string_very_slow {
...
     # when the worker host is utilized so better wait until the string is
     # displayed before continuing
-    wait_still_screen 1;
+    # For the special winter grub screen with moving penguins
+    # `wait_still_screen` does not work so we just revert to sleeping a bit
+    # instead of waiting for a still screen which is never happening. Sleeping
+    # for 3 seconds is less waste of time than waiting for the
+    # wait_still_screen to timeout, especially because wait_still_screen is
+    # also scaled by TIMEOUT_SCALE which we do not need here.
+    if (get_var('WINTER_IS_COMING', '')) {
```
Comment (coolo) [inline lib/utils.pm]: you don't need a default empty string here to pass as false. You only need one if you compare it with strings

File: products/opensuse/main.pm [hunk 2]
```diff
@@ -78,10 +78,13 @@ sub cleanup_needles() {
...
     # unregister christmas needles unless it is December where they should
     # appear. Unused needles should be disregarded by admin delete then
-    my @time = localtime();
-    unregister_needle_tags('CHRISTMAS') unless $time[4] == 11;
+    unregister_needle_tags('CHRISTMAS') unless get_var('WINTER_IS_COMING', '');
 }
 
+# we need some special handling for the openSUSE christmas needles
+my @time = localtime();
+set_var('WINTER_IS_COMING', 1) if $time[4] == 11;
```
Comment (coolo) [inline products/opensuse/main.pm]: this calculation is unfortunately off. the actual theme code defines the likeness of the winter theme as:

```
 over month 12 eq { 5 add } if
  over month 1 le { 5 add } if
  over weekday 5 eq { 2 mul } if
  over weekday 6 eq { 4 mul } if
  over dup month 12 eq exch day dup 24 ge exch 6 eq or and { pop 100 } if
  over dup month 1 eq exch day 1 eq and { pop 100 } if
```

So you will have to worry about WINTER in january as well - which is actually not coming at that point, but at its high time :)

Comment (coolo) [inline products/opensuse/main.pm]: outside of december and january the chance is 0, yes

Comment (coolo) [inline products/opensuse/main.pm]: today it's 100% :)

**Comment by okurz** [comment]
updated

**Pull request merged**